### PR TITLE
feat(Decorator design pattern):[20-Jan-24] Decorator design pattern

### PR DESCRIPTION
**Pull request by FarhanAhmadISM**

### CODE REVIEW

**Comment by FarhanAhmadISM** [comment]
+testpr

File: DecoratorDesignPattern/Decorator/MushroomTopping.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+public class MushroomTopping extends ToppingDecorator {
+    BasePizza pizza;
```
Comment (sunilagwl5) [inline DecoratorDesignPattern/Decorator/MushroomTopping.java:2]: Use Encapsulation for data hiding..

File: DecoratorDesignPattern/Decorator/ExtraCheeseTopping.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+public class ExtraCheeseTopping extends ToppingDecorator {
+    BasePizza pizza;
+    ExtraCheeseTopping(BasePizza pizza) {
+        this.pizza = pizza;
+    }
+
```
Comment (sunilagwl5) [inline DecoratorDesignPattern/Decorator/ExtraCheeseTopping.java:6]: we can remove code duplication and code readability by

moving this logic inside TopicDecorator and use as -

`ExtraCheeseTopping(BasePizza pizza) {
        super(pizza)
    }`

File: DecoratorDesignPattern/Pizza/BasePizza.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+public abstract class BasePizza {
+    public abstract int cost();
```
Comment (sunilagwl5) [inline DecoratorDesignPattern/Pizza/BasePizza.java:2]: Let's Use interface.

**Pull request merged**